### PR TITLE
added alert and dark mode

### PR DESCRIPTION
**Pull request by Nilesh1308**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,31 +3,50 @@ import { useState } from 'react';
...
-      }
-  }
-  const showData = () =>{
-  axios.get('https://gorest.co.in/public/v2/users')
-  .then((res)=> setMyData(res.data));
+      document.body.style.backgroundColor ='white';
+      showAlert("Light mode has been enabled","success");
+    }
   }
+  //const showData = () =>{
```
Comment (lordsonvimal) [inline src/App.js]: You can remove the commented lines if you are not planning to use it immediately. Keeping the code clean will help you in maintaining it.

File: src/App.js [hunk 2]
```diff
@@ -3,31 +3,50 @@ import { useState } from 'react';
...
+  const [mode, setMode] = useState('light'); //wheather dark mode is enabled or not
+ // const [myData, setMyData] = useState([])
+ const [alert, setAlert] = useState(null)
+
+ const showAlert =(message,type)=>{
+  setAlert({
+    msg:message,
+    type:type
+  })
+  setTimeout(() => {
```
Comment (lordsonvimal) [inline src/App.js:26]: 👍 

I hope now you understand why you need to use `setTimeout` and when to use it.

File: src/App.js (hunk 1 above)
Comment (Nilesh1308) [reply src/App.js]: commented code is logesh's task it was on this app only

File: src/components/Alert1.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+function Alert1(props) {
```
Comment (lordsonvimal) [inline src/components/Alert1.js:3]:
```suggestion
function Alert(props) {
```

The filename can be `Alert` as well

Comment (Nilesh1308) [reply src/components/Alert1.js:3]: i was trying Alert but it was showing me error while i was importing it

File: src/components/Textform.js [hunk 4]
```diff
@@ -6,42 +6,46 @@ export default function Textform(props) {
     //console.log("uppercase was clicked"+ text);
     let newText= text.toUpperCase();
     setText(newText);
+    props.showAlert("Converted to uppercase","success");
```
Comment (lordsonvimal) [inline src/components/Textform.js]: You can destructure the props like `const { showAlert } = props;` at the top of the function.

File: src/App.js (hunk 2 above)
Comment (Nilesh1308) [reply src/App.js:26]: yes used it once but got to know about setTimeout

File: src/components/Textform.js [hunk 5]
```diff
@@ -6,42 +6,46 @@ export default function Textform(props) {
...
   }
   const[text, setText] = useState(''); 
   return (
     <>
-    <div className="container">
+    <div className="container"style={{color: props.mode==='dark'?'white':'#042743'}}> 
             <h2>{props.heading}</h2>
         <div className="mb-3">
-        <textarea className="form-control" value ={text} onChange={handleOnChange} id="myBox"  rows="8"></textarea>
+        <textarea className="form-control" value ={text} onChange={handleOnChange}  id="myBox"  rows="8" style={{backgroundColor: props.mode==='dark'?'grey':'white',color: props.mode==='dark'?'white':'#042743'}}></textarea>
```
Comment (lordsonvimal) [inline src/components/Textform.js:36]: Try using css classnames instead of direct styles

**Pull request merged**